compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1850,6 +1865,46 @@ pub(super) fn compare_impl_const_raw(
         return Err(diag.emit());
     };
 
+    let impl_ct_predicates = tcx.predicates_of(impl_ct.def_id);
+    let trait_ct_predicates = tcx.predicates_of(trait_ct.def_id);
+
+    // FIXME(generic_consts): Is this correct? We already have an ocx...
+    check_region_bounds_on_impl_item(tcx, impl_ct, trait_ct, false)?;
+
+    // FIXME(generic_consts): It'd be nice if we added a fast-path if the own bounds are empty
+    // similar to cmp_impl_ty_pred_entail without intro'ing unsoundness.
+    let impl_ct_own_bounds = impl_ct_predicates.instantiate_own(tcx, impl_substs);
+
+    // The predicates declared by the impl definition, the trait and the
+    // associated const in the trait are assumed.
+    let impl_predicates = tcx.predicates_of(impl_ct_predicates.parent.unwrap());
+    let mut hybrid_preds = impl_predicates.instantiate_identity(tcx);
+    hybrid_preds.predicates.extend(
+        trait_ct_predicates
+            .instantiate_own(tcx, trait_to_impl_substs)
+            .map(|(predicate, _)| predicate),
+    );
+
+    let hybrid_param_env = ty::ParamEnv::new(
+        tcx.mk_clauses(&hybrid_preds.predicates),
+        Reveal::UserFacing,
+        hir::Constness::Const,
+    );
+    let hybrid_param_env = traits::normalize_param_env_or_error(
+        tcx,
+        hybrid_param_env,
+        ObligationCause::misc(impl_ct_span, impl_ct_def_id),
+    );
+
+    // FIXME(generic_consts): Is this correct? We already have an ocx...
```

Comment:
It's kinda sketchy, since outlives that come from solving these obligations are using the `outlives_env` from a different param-env. IMO everything should just be done in the hybrid-env, like how `compare_method_predicate_entailment` does it (I think).